### ISSUE TITLE
V36 fails to run openSUSE 13.2

### ISSUE THREAD

**Issue by theoriginalbigbaddave** [micheleg/dash-to-dock#131]
After upgrade to v36 the extension fails to run.

I'm running openSUSE 13.2 X86_64 and GNOME shell 3.14.1.

The error recorded in the systemd journal and lg is: TypeError: this.app is undefined

I tried: manual re-install (from zip file), disabling all other extensions, resetting dconf settings to defaults and logging out and back in but the issue persisted.

v35 works ok.

![screenshot from 2015-01-09 16 42 05](https://cloud.githubusercontent.com/assets/1520025/5683492/46c90862-9820-11e4-9dfe-7c9c79fe194e.png)
![screenshot from 2015-01-09 16 39 51](https://cloud.githubusercontent.com/assets/1520025/5683499/507280be-9820-11e4-9efa-b780ad4dcf71.png)

**Comment by micheleg** [comment]
Hi. Would you mind checking the content of the directory

```
~/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/
```

 and its permissions? Can you confirm that the problem is reproducible? You can  use the makefile from the git source or manually extract the content of the zip file to that directory and then try enabling the extension with GNOME Tweak tool if nothing else work.

**Comment by micheleg** [comment]
Also, could you run in a terminal
```
gnome-shell --replace
```
and the try reinstalling the extension and then look for errors in that terminal?

**Comment by theoriginalbigbaddave** [comment]
ls -l .local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com
total 188
-rw-r--r-- 1 dave users  4088 Jan  5 19:38 convenience.js
-rw-r--r-- 1 dave users 17993 Jan  5 19:38 COPYING
-rw-r--r-- 1 dave users 52783 Jan  5 19:38 dockedDash.js
-rw-r--r-- 1 dave users  1487 Jan  5 19:38 extension.js
-rw-r--r-- 1 dave users 10851 Jan  5 19:38 intellihide.js
drwxr-xr-x 1 dave users    48 Jan 10 18:30 locale/
drwxr-xr-x 1 dave users   274 Jan  5 19:38 media/
-rw------- 1 dave users   822 Jan  8 11:39 metadata.json
-rw-r--r-- 1 dave users 47322 Jan  5 19:38 myDash.js
-rw-r--r-- 1 dave users 30165 Jan  5 19:38 prefs.js
-rw-r--r-- 1 dave users  1850 Jan  5 19:38 README.md
drwxr-xr-x 1 dave users   136 Jan 10 18:30 schemas/
-rw-r--r-- 1 dave users  4644 Jan  5 19:38 stylesheet.css

**Comment by theoriginalbigbaddave** [comment]
GNOME Shell started at Sat Jan 10 2015 18:29:49 GMT+0000 (GMT)
Archive:  /tmp/0UJLSX.shell-extension.zip
  inflating: /home/dave/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/schemas/gschemas.compiled  
   creating: /home/dave/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/zh_CN/LC_MESSAGES/
   creating: /home/dave/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/pt/
   creating: /home/dave/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/media/
   creating: /home/dave/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/pl/
   creating: /home/dave/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/ja/LC_MESSAGES/
  inflating: /home/dave/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/media/four_rtl.svg  
   creating: /home/dave/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/pt_BR/LC_MESSAGES/
  inflating: /home/dave/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/media/two.svg  
  inflating: /home/dave/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/README.md  
   creating: /home/dave/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/cs/LC_MESSAGES/
  inflating: /home/dave/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/media/one.svg  
  inflating: /home/dave/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/pt/LC_MESSAGES/dashtodock.mo  
  inflating: /home/dave/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/de/LC_MESSAGES/dashtodock.mo  
  inflating: /home/dave/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/convenience.js  
  inflating: /home/dave/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/cs/LC_MESSAGES/dashtodock.mo  
  inflating: /home/dave/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/media/two_bottom.svg  
  inflating: /home/dave/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/media/three_bottom.svg  
  inflating: /home/dave/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/ja/LC_MESSAGES/dashtodock.mo  
  inflating: /home/dave/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/fr/LC_MESSAGES/dashtodock.mo  
  inflating: /home/dave/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/media/three.svg  
  inflating: /home/dave/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/media/one_rtl.svg  
  inflating: /home/dave/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/stylesheet.css  
   creating: /home/dave/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/hu/LC_MESSAGES/
  inflating: /home/dave/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/media/one_bottom.svg  
  inflating: /home/dave/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/media/two_rtl.svg  
  inflating: /home/dave/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/hu/LC_MESSAGES/dashtodock.mo  
   creating: /home/dave/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/pl/LC_MESSAGES/
  inflating: /home/dave/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/media/four_bottom.svg  
  inflating: /home/dave/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/myDash.js  
  inflating: /home/dave/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/media/three_rtl.svg  
  inflating: /home/dave/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/extension.js  
  inflating: /home/dave/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/dockedDash.js  
  inflating: /home/dave/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/zh_CN/LC_MESSAGES/dashtodock.mo  
  inflating: /home/dave/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/schemas/org.gnome.shell.extensions.dash-to-dock.gschema.xml  
  inflating: /home/dave/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/intellihide.js  
  inflating: /home/dave/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/pl/LC_MESSAGES/dashtodock.mo  
  inflating: /home/dave/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/pt_BR/LC_MESSAGES/dashtodock.mo  
  inflating: /home/dave/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/COPYING  
  inflating: /home/dave/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/media/four.svg  
  inflating: /home/dave/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/prefs.js  
 extracting: /home/dave/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/metadata.json  
Gjs-Message: JS LOG: Extension "dash-to-dock@micxgx.gmail.com" had error: TypeError: this.app is undefined

(gnome-shell:7669): Gjs-WARNING **: JS ERROR: TypeError: this.app is undefined
AppIcon<._onStateChanged@resource:///org/gnome/shell/ui/appDisplay.js:1592
wrapper@resource:///org/gnome/gjs/modules/lang.js:169
myAppIcon<._updateRunningStyle@/home/dave/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/myDash.js:1133
wrapper@resource:///org/gnome/gjs/modules/lang.js:169
myAppIcon<._init@/home/dave/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/myDash.js:1110
wrapper@resource:///org/gnome/gjs/modules/lang.js:169
_Base.prototype._construct@resource:///org/gnome/gjs/modules/lang.js:110
Class.prototype._construct/newClass@resource:///org/gnome/gjs/modules/lang.js:204
myDash<._createAppItem@/home/dave/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/myDash.js:517
wrapper@resource:///org/gnome/gjs/modules/lang.js:169
myDash<._redisplay@/home/dave/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/myDash.js:768
wrapper@resource:///org/gnome/gjs/modules/lang.js:169
_runDeferredWork@resource:///org/gnome/shell/ui/main.js:523
_runAllDeferredWork@resource:///org/gnome/shell/ui/main.js:532
queueDeferredWork/_deferredTimeoutId<@resource:///org/gnome/shell/ui/main.js:614

**Comment by micheleg** [comment]
Can you try replacing line 1133 of file myDash.js
```
AppDisplay.AppIcon.prototype._onStateChanged();
```
to
```
AppDisplay.AppIcon.prototype._onStateChanged.call(this);
```
 and then reloading the shell?

I think this should solve your problem

**Comment by theoriginalbigbaddave** [comment]
Spot on that fixed it.

**Comment by micheleg** [comment]
merged 4006b2d0.
